package/boot/uboot-envtools/files/mvebu

```diff
@@ -32,6 +32,9 @@ marvell,armada8040-mcbin-doubleshot|\
 marvell,armada8040-mcbin-singleshot)
 	ubootenv_add_uci_config "/dev/mtd0" "0x3f0000" "0x10000" "0x10000" "1"
 	;;
+globalscale,espressobin-ultra)
+	ubootenv_add_uci_config "/dev/mtd2" "0x3f0000" "0x10000" "0x10000" "1"
```

Comment:
Did you test this? The dts you're adding has:
```
			partition@0 {
				label = "u-boot";
				reg = <0x0 0x3e0000>;
			};
			partition@3e0000 {
				label = "hw-info";
				reg = <0x3e0000 0x10000>;
				read-only;
			};
			partition@3f0000 {
				label = "u-boot-env";
				reg = <0x3f0000 0x10000>;
			};
```

So that cannot work.

```suggestion
	idx="$(find_mtd_index u-boot-env)"
	[ -n "$idx" ] && \
		ubootenv_add_uci_config "/dev/mtd$idx" "0x0" "0x10000" "0x10000" "1"
```

Reply:
I did, not sure why it would not work. This is the output from the device I have:

```
root@OpenWrt:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 003e0000 00010000 "u-boot"
mtd1: 00010000 00010000 "hw-info"
mtd2: 00010000 00010000 "u-boot-env"

root@OpenWrt:~# cat /etc/fw_env.config
/dev/mtd2 0x3f0000 0x10000 0x10000 1

root@OpenWrt:~# fw_printenv
arch=arm
baudrate=115200
board=mvebu_armada-37xx
board_name=mvebu_armada-37xx
bootargs=console=ttyMV0,115200 earlycon=ar3700_uart,0xd0012000
bootcmd=mmc dev 0; load mmc 0:1 $loadaddr boot.scr; source $loadaddr
bootcmd_nfs=run get_images; run set_bootargs; booti $kernel_addr_r $ramdisk_addr_r $fdt_addr_r
bootdelay=2
console=console=ttyMV0,115200 earlycon=ar3700_uart,0xd0012000
cpu=armv8
...
```

Reply:
Huh, 0x3f0000 is the offset, and none of those 3 partitions is big enough.
I got an error in that situation, which is why I added the suggested block in another PR, see #3405.
Can you please try my suggestion? That block is used by various boards already.

Reply:
both cases work for me. I see that #3405 is still open, so I can remove this part for now and sent new PR just for this once your PR is merged. Would that be okay with you?

Reply:
Sure, but that makes it a little incomplete, no? If you're fine with the suggested change, why not use it. This and the #3405 case would still be different blocks due to the fallback in my PR, so no need to wait for it.

Reply:
@dhewg okay, opted with you suggestion and updated PR. I would appreciate if you could add espressobin-ultra under the globalscale boards case in #3405 if this one gets merged before, so that everything is unified there.

Reply:
Thanks, and sounds like a plan, will do! :)

Reply:
sorry, just now realized that I didn't push the change in the last PR. Fixed now.

Reply:
;) Thanks!

Reply:
#3405 has been merged, you can now just add the board name to the existing `case`